test/components/tableTests.ts

```diff
@@ -294,77 +118,346 @@ describe("Tables", () => {
     let c5 = new Plottable.Component();
     let c6 = new Plottable.Component();
     let table: Plottable.Components.Table;
-    it("works in basic case", () => {
+
+    it("removes the specified Component", () => {
       table = new Plottable.Components.Table([[c1, c2], [c3, c4], [c5, c6]]);
       table.remove(c4);
-      assert.deepEqual((<any> table)._rows, [[c1, c2], [c3, null], [c5, c6]], "remove one element");
+      assertTableRows(table, [[c1, c2], [c3, null], [c5, c6]], "the requested element was removed");
+      assert.isNull(c4.parent(), "Component disconnected from the Table");
     });
 
     it("does nothing when component is not found", () => {
       table = new Plottable.Components.Table([[c1, c2], [c3, c4]]);
       table.remove(c5);
-
-      assert.deepEqual((<any> table)._rows, [[c1, c2], [c3, c4]], "remove nonexistent component");
+      assertTableRows(table, [[c1, c2], [c3, c4]], "removing a nonexistent Component does not affect the table");
     });
 
-    it("removing component twice should have same effect as removing it once", () => {
+    it("has no further effect when called a second time with the same Component", () => {
       table = new Plottable.Components.Table([[c1, c2, c3], [c4, c5, c6]]);
 
-      table.remove(c1);
-      assert.deepEqual((<any> table)._rows, [[null, c2, c3], [c4, c5, c6]], "item twice");
+      let expectedRows = [[null, c2, c3], [c4, c5, c6]];
```

Comment:
Can we factor out the variables and have `expectedRows` depend on the input?  Main concern is that let's say that someone were to add a new row to the input just to check behavior.  Then the test would fail unless that person changes `expectedRows` as well.